### ISSUE TITLE
BadRequest400Exception: mapper [created_at] of different type, current_type [text], merged_type [date]

### ISSUE THREAD

**Issue by sebestenyb** [babenkoivan/scout-elasticsearch-driver#92]
Hi,

I'm trying to run a zero downtime migration `php artisan elastic:migrate App\\MyModel my_index_v2` after adding a mapping of `date` to the `created_at` field of the model, and ran into the following error:

```
The my_index_v2 index was created.
The my_index_v2 mapping was updated.
The my_index_v2_write alias for the my_index_v2 index was created.

Elasticsearch\Common\Exceptions\BadRequest400Exception]                                                                                                                                                                                                                                               
  {"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"mapper [created_at] of different type, current_type [text], merged_type [date]"}],"type":"illegal_argument_exception","reason":"mapper [created_at] of different type, current_type [text], merged_type [date]"},"status":400} 
```

The new index shows the correct mapping:

```
{
    "my_index_v2": {
        "mappings": {
            "my_doc": {
                "properties": {
                    "created_at": {
                        "type": "date"
                    },
...
```

Am I doing something wrong here? 
I thought the point of the migrate command is that it creates a new index with the new mapping, and reindexes all the documents into it.
But from this error message it seems the new index somehow inherits the old mapping before trying to change it.

ElasticSearch version is **6.2.2**


Thanks, Sebi

**Comment by babenkoivan** [comment]
Hi @sebestenyb,

It should work exactly as you described. 
It looks like an a bug, I'll have a look.

**Comment by babenkoivan** [comment]
Hi again,

I made a fix in `v3.7.1`. Please, update the package and give it a try.

**Comment by sebestenyb** [comment]
Ah great work!

Just noticed that somewhere along the line the package's PHP requirement has been bumped to 7.1, but I don't see any code changes really needing that.
I think it could getaway with `"laravel/scout": ">=3.0"`.

Keep up the good work!

Thanks, Sebi
